File: android/app/src/main/java/ai/thetahealth/mirobody/ui/AppErrorMessage.kt
package ai.thetahealth.mirobody.ui

import android.content.Context
import ai.thetahealth.mirobody.R
import ai.thetahealth.mirobody.data.net.AppError

/** Render an [AppError] as a localized, user-friendly string using the given context. */
fun AppError.toLocalizedMessage(context: Context): String = when (this) {
    is AppError.Auth -> context.getString(R.string.error_auth)
    is AppError.Network -> context.getString(R.string.error_network)
    is AppError.Http -> context.getString(R.string.error_http, status)
    is AppError.Api -> {
        val suffix = serverMessage?.takeIf { it.isNotBlank() }?.let { ": $it" } ?: ""
        context.getString(R.string.error_api, code, suffix)
    }
    is AppError.Parse -> context.getString(R.string.error_parse)
    is AppError.Unknown -> context.getString(R.string.error_unknown)
}

File: ios/Mirobody/UI/AppErrorMessage.swift
import Foundation

/// Renders an `AppError` as a localized, user-facing string. Mirrors
/// `ui/AppErrorMessage.kt`.
func localizedMessage(_ error: AppError, language: String) -> String {
    switch error {
    case .auth:
        return L("error_auth", language)
    case .network:
        return L("error_network", language)
    case .http(let status, _):
        return L("error_http", language, status)
    case .api(let code, let serverMessage, _):
        let suffix = serverMessage?.nonBlank.map { ": \($0)" } ?? ""
        return L("error_api", language, code, suffix)
    case .parse:
        return L("error_parse", language)
    case .unknown:
        return L("error_unknown", language)
    }
}
